### PR TITLE
chore: added tests for role access

### LINKED ISSUE #45: Tests for Role based access
Tests for Role based access 
------
**Description:** 
Ensure that all permissioned functions are under role enforcement in one of the following three roles: 
- DEBT_MANAGER: updateDebt and setters
- STRATEGY_MANAGER: add remove and migrate strategies and their setters
- ACCOUNTING_MANAGER: process reports

Also, ensure that the governance role can do role changes successfully. 

**Dependencies:**
Bounty #44  is done

**Acceptance criteria:** 
All tests are implemented and running. 
¡
**Payment Trigger:** 
PR is merged
**Category:**
DEV
**Level:**
1

### PR DESCRIPTION
**Pull request by bufander**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #45 

## Checklist

- [x] I have run vyper and solidity linting
- [x] I have run the tests on my machine
- [x] I have followed commitlint guidelines
- [x] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

File: tests/unit/vault/test_role_base_access.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+    with ape.reverts():
+        vault.update_debt(strategy, sender=bunny)
+
+
+def test_update_debt__debt_manager(
+    gov, mint_and_deposit_into_vault, vault, strategy, bunny
+):
+
+    # We temporaly give bunny the role of DEBT_MANAGER
```
[the remark this answers is not in the document]
Comment (bufander) [reply tests/unit/vault/test_role_base_access.py:125]: Done in next commit

File: tests/unit/vault/test_role_base_access.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    new_strategy = create_strategy(vault)
+    with ape.reverts():
+        vault.migrate_strategy(strategy, new_strategy, sender=bunny)
+
+
+def test_migrate_strategy__strategy_manager(
+    vault, strategy, create_strategy, gov, bunny
+):
+
+    # We temporaly give bunny the role of STRATEGY_MANAGER
```
[the remark this answers is not in the document]
Comment (bufander) [reply tests/unit/vault/test_role_base_access.py:56]: Done in next commit

File: tests/unit/vault/test_role_base_access.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+
+def test_revoke_strategy__no_strategy_manager__reverts(vault, strategy, bunny):
+    with ape.reverts():
+        vault.revoke_strategy(strategy, sender=bunny)
+
+
+def test_revoke_strategy__strategy_manager(vault, strategy, gov, bunny):
+
+    # We temporaly give bunny the role of STRATEGY_MANAGER
```
[the remark this answers is not in the document]
Comment (bufander) [reply tests/unit/vault/test_role_base_access.py:35]: Done in next commit

File: tests/unit/vault/test_role_base_access.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    max_debt_for_strategy = 1
+    with ape.reverts():
+        vault.update_max_debt_for_strategy(
+            strategy, max_debt_for_strategy, sender=bunny
+        )
+
+
+def test_update_max_debt__debt_manager(gov, vault, strategy, bunny):
+
+    # We temporaly give bunny the role of DEBT_MANAGER
```
[the remark this answers is not in the document]
Comment (bufander) [reply tests/unit/vault/test_role_base_access.py:106]: Done in next commit

File: tests/unit/vault/test_role_base_access.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+def test_set_minimum_total_idle__no_debt_manager__reverts(bunny, vault):
+
+    minimum_total_idle = 1
+    with ape.reverts():
+        vault.set_minimum_total_idle(minimum_total_idle, sender=bunny)
+
+
+def test_set_minimum_total_idle__debt_manager(gov, vault, bunny):
+
+    # We temporaly give bunny the role of DEBT_MANAGER
```
[the remark this answers is not in the document]
Comment (bufander) [reply tests/unit/vault/test_role_base_access.py:85]: Done in next commit

File: tests/unit/vault/test_role_base_access.py [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+
+def test_add_strategy__no_strategy_manager__reverts(vault, create_strategy, bunny):
+    new_strategy = create_strategy(vault)
+    with ape.reverts():
+        vault.add_strategy(new_strategy, sender=bunny)
+
+
+def test_add_strategy__strategy_manager(vault, create_strategy, gov, bunny):
+
```
[the remark this answers is not in the document]
Comment (bufander) [reply tests/unit/vault/test_role_base_access.py:17]: yes. Linting on Github CI has also passed
image


what do you see that looks wrong?

**Comment by anticlimactic** [comment]
there should be additional tests for each fn that test the for the absence of the role (i.e. try to call the fn with bunny before you grant the role to bunny).

**Comment by bufander** [comment]
> there should be additional tests for each fn that test the for the absence of the role (i.e. try to call the fn with bunny before you grant the role to bunny).

don´t you think we are checking that on the above test where we provide a temporary role? 

(i.e. we check that bunny is unable to call `add_strategy` due to the absence of role on test `test_add_strategy__no_strategy_manager__reverts`, then we provide temporarily role and we check that is able to call `add_strategy` on test `test_add_strategy__strategy_manager`)



Maybe I am missing smth

**Pull request merged**